### DISCUSSION TITLE
Calculate the time duration from start to end

### DISCUSSION THREAD

**Discussion by BismaIftekhar** [Budibase/budibase#12558] [Help] [answered]
Hello Budibase community,

I am trying to calculate the time difference from the start date shown by the "startdatum" field and start time "startzeit" to the End date and endtime field. So, it should show the the total time in days/hours from start of date&time to end of date&time. The date format is dd.MM.yyyy and time field is text with hh:mm format.

![Budibase_timediff](https://github.com/Budibase/budibase/assets/96243721/61dc9af3-fe4b-4ed8-b4c4-f5bf280627d1)
Can anyone please explain how can I do this?

Thank you so much.

**Comment by melohagan** [comment, marked as answer]
Hey @BismaIftekhar 

There's a guide here that should get you started! https://docs.budibase.com/docs/update-date-field-on-change

**Reply by BismaIftekhar** [reply]
Thank you for your prompt response. The link is helful but it does not take into account the start time and end time selected for those days. Is this possible to take that into account too?when calculating the difference.

**Reply by melohagan** [reply]
Yes it would be possible with JavaScript, you may need to do some searching on the internet to figure out the exact code needed to get the difference you are looking for!

The tutorial however will give you the Budibase principles required.
